File: frontend/src/types/tag.ts
export type TagType = 'knowledge_point' | 'problem_type' | 'method' | 'mistake_reason'

export interface TagItem {
  tag_id: number
  tag_name: string
  tag_type: TagType
  usage_count: number
  is_active: boolean
}

export interface TagListResponse {
  list: TagItem[]
}

export interface CreateTagPayload {
  tag_name: string
  tag_type: TagType
}

File: backend/internal/domain/model/tag.go
package model

import "time"

type Tag struct {
	ID         int64
	UserID     int64
	TagName    string
	TagType    string
	UsageCount int
	IsActive   bool
	DeletedAt  *time.Time
	CreatedAt  time.Time
	UpdatedAt  time.Time
}

type TagCollections struct {
	KnowledgePoints []string
	ProblemType     []string
	Method          []string
	MistakeReason   []string
}
